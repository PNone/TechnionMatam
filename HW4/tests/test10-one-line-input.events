PotionsMerchant PotionsMerchant SolarEclipse Pack 3 Snail Slime Slime Slime Slime Balrog Pack 2 Slime Snail SolarEclipse