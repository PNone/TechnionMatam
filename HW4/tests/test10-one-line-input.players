Daniel Warrior Responsible Regev Magician Responsible Braa Magician RiskTaking