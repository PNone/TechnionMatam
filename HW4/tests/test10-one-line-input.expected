Game is about to start!
These are the players:

1. Daniel, Warrior with Responsible character (level 1, force 5)
2. Regev, Magician with Responsible character (level 1, force 5)
3. Braa, Magician with RiskTaking character (level 1, force 5)

----------------------------------------

New round starts!

Turn: 1
Player: Daniel, Warrior with Responsible character (level 1, force 5)
Event: PotionsMerchant
Daniel bought 0 potions!

Turn: 2
Player: Regev, Magician with Responsible character (level 1, force 5)
Event: PotionsMerchant
Regev bought 0 potions!

Turn: 3
Player: Braa, Magician with RiskTaking character (level 1, force 5)
Event: SolarEclipse
Braa was affected by a solar eclipse! their force has changed by 1!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         6     100   10
2          Daniel          1         5     150   10
3          Regev           1         5     100   10

----------------------------------------

New round starts!

Turn: 4
Player: Daniel, Warrior with Responsible character (level 1, force 5)
Event: Pack of 3 members (power 29, loot 12, damage 60)
Daniel lost the encounter and took 60 damage!

Turn: 5
Player: Regev, Magician with Responsible character (level 1, force 5)
Event: Slime (power 12, loot 5, damage 25)
Regev lost the encounter and took 25 damage!

Turn: 6
Player: Braa, Magician with RiskTaking character (level 1, force 6)
Event: Slime (power 12, loot 5, damage 25)
Braa lost the encounter and took 25 damage!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         6     75    10
2          Daniel          1         5     80    10
3          Regev           1         5     75    10

----------------------------------------

New round starts!

Turn: 7
Player: Daniel, Warrior with Responsible character (level 1, force 5)
Event: Balrog (power 15, loot 100, damage 9001)
Daniel lost the encounter and took 9001 damage!

Turn: 8
Player: Regev, Magician with Responsible character (level 1, force 5)
Event: Pack of 2 members (power 17, loot 7, damage 35)
Regev lost the encounter and took 35 damage!

Turn: 9
Player: Braa, Magician with RiskTaking character (level 1, force 6)
Event: SolarEclipse
Braa was affected by a solar eclipse! their force has changed by 1!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         7     75    10
2          Daniel          1         5     0     10
3          Regev           1         5     40    10

----------------------------------------

New round starts!

Turn: 10
Player: Regev, Magician with Responsible character (level 1, force 5)
Event: PotionsMerchant
Regev bought 2 potions!

Turn: 11
Player: Braa, Magician with RiskTaking character (level 1, force 7)
Event: PotionsMerchant
Braa bought 0 potions!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         7     75    10
2          Daniel          1         5     0     10
3          Regev           1         5     60    0

----------------------------------------

New round starts!

Turn: 12
Player: Regev, Magician with Responsible character (level 1, force 5)
Event: SolarEclipse
Regev was affected by a solar eclipse! their force has changed by 1!

Turn: 13
Player: Braa, Magician with RiskTaking character (level 1, force 7)
Event: Pack of 3 members (power 29, loot 12, damage 60)
Braa lost the encounter and took 60 damage!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         7     15    10
2          Daniel          1         5     0     10
3          Regev           1         6     60    0

----------------------------------------

New round starts!

Turn: 14
Player: Regev, Magician with Responsible character (level 1, force 6)
Event: Slime (power 12, loot 5, damage 25)
Regev lost the encounter and took 25 damage!

Turn: 15
Player: Braa, Magician with RiskTaking character (level 1, force 7)
Event: Slime (power 12, loot 5, damage 25)
Braa lost the encounter and took 25 damage!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         7     0     10
2          Daniel          1         5     0     10
3          Regev           1         6     35    0

----------------------------------------

New round starts!

Turn: 16
Player: Regev, Magician with Responsible character (level 1, force 6)
Event: Balrog (power 17, loot 100, damage 9001)
Regev lost the encounter and took 9001 damage!

Round has ended!
Leader board is as follows:
Ranking    Player Name     Level     Force HP    Coins
1          Braa            1         7     0     10
2          Daniel          1         5     0     10
3          Regev           1         6     0     0

----------------------------------------

Game is over!
There are no winners in this game.
Better luck next time!
